### DISCUSSION TITLE
Use direct imports over asterisk?

### DISCUSSION THREAD

**Discussion by MickL** [open-circle/valibot#967] [Q&A]
In the docs the imports are always using asterik:

```ts
import * as v from 'valibot';
```

But wouldnt it be better to import every function separately? Or does it make no difference for tree shaking?

```ts
import { parse, pipe, maxLength, object, string } from 'valibot';
```

To me the first example sounds like "import the whole goddamn thing like back in the old days", while the second one is "ah yes, give me just what I need"

**Comment by fabian-hiller** [comment]
It makes no difference for tree shaking and provides a better DX in most cases. That's why we recommend using a wildcard import.
